src/libcrun/seccomp.c

```diff
@@ -82,6 +82,12 @@
 #  define SECCOMP_FILTER_FLAG_WAIT_KILLABLE_RECV (1UL << 5)
 #endif
 
+#define SECCOMP_CACHE_DIR ".seccomp-cache"
+
+/* Heuristic to avoid the cache directory grows indefinitely.  The inode size for
+   the directory is proportional to the number of entries it contains.  */
+#define MAX_CACHE_DIR_INODE_SIZE 1024
```

Comment:
I did an experiment, and on a tmpfs the size increased by 20 each time a file was added. That makes this 50 files.
I have no idea how common the use of different seccomp rules are though, so i have no idea what would make a good cache size.

Reply:
I couldn't find a better (and still cheap) condition to trigger the cleanup.

I thought about using `$RUNDIR/crun/.seccomp-cache/$DIGEST/data` instead of `$RUNDIR/crun/.seccomp-cache/$DIGEST` so that we could see immediately how many entries there are in .seccomp-cache looking at the st_nlink, but it is a bit more difficult to handle since all the atomic operations would require rename/renameat tricks.

Reply:
Its probably fine, I was mostly worried about whether this is a good default size or not.

Reply:
another option could be to check how much bigger that `$RUNDIR` it is.  Since the `$RUNDIR` size depends on the number of active containers, something like `threshold = max(MAX_CACHE_DIR_INODE_SIZE, $RUNDIR.st_size * 3)`

Reply:
tweaked to:

```c
#define MIN_CACHE_DIR_INODE_SIZE 1024

static inline off_t
get_cache_dir_inode_max_size (off_t size_rundir)
{
  return size_rundir * 3 + MIN_CACHE_DIR_INODE_SIZE;
}
```

so we account for the number of containers as well.